### PR TITLE
Edit Unavailability Endpoint

### PR DESCRIPTION
**Pull request by TaiHaDev**
## Describe your changes
The logic for updating unavailability is simple. Updating any field in the unavailability with the matching fields in the payload. The most important part is the payload validation, the payload data is checked in the repository function and raise custom errors to catch later in the controller. Some features:
- a user can only edit their own unavailability (this feature will be later discussed within the meeting to see if it should be applied for other API endpoints)
- custom errors (EventNotFoundError, InvalidArgumentError)

### CODE REVIEW

File: controllers/v2/unavailability/api.py [hunk 1]
```diff
@@ -20,16 +21,25 @@ def __init__(self, event_repository: EventRepository = EventRepository()):
         self.event_repository = event_repository
 
     @requires_auth
-    @is_user_or_has_role(None, UserType.ROOT_ADMIN)
     def put(self, user_id, event_id):
-        args = edit_parser.parse_args()
-        success = self.event_repository.edit_event(user_id, event_id, **args)
-        if success is True:
+        @is_user_or_has_role(user_id, UserType.ROOT_ADMIN)
```
Comment (BenMMcLean) [inline controllers/v2/unavailability/api.py:25]: Why move this check to here? IIRC the decorator should automatically look for a "user_id" path parameter. This kind of setup should only be necessary if the relevant user_id isn't in the path.

**Review by BenMMcLean — COMMENTED** [review]
Good work! Just looking for some explanation about the choice to move the is_user_or_has_role check before approval.

File: controllers/v2/unavailability/api.py (hunk 1 above)
Comment (TaiHaDev) [reply controllers/v2/unavailability/api.py]: Hi Ben, I changed it back. You are correct!  I just have a little bit of misunderstanding. Thanks :-)

**Review by BenMMcLean — APPROVED** [review]
Good job!

**Review by naychithanshwe — APPROVED** [review]

**Pull request merged**